### PR TITLE
(#28) Clinical Trials and Thesaurus links

### LINKED ISSUE #28: Story: Clinical Trials/NCI Thesaurus Links on Term Page
## As a user, I would like to find active clinical trials and learn more about a drug from the NCI Thesaurus so that I may be more informed about the cancer drug and my trial options.

> **ESTIMATE** 

### Acceptance Criteria

```gherkin
Scenario: As a user, I would like to see a link to active clinical trials when I am viewing a term page
  Given "dynamicListingPageBase" is set to "https://www.cancer.gov/about-cancer/treatment/clinical-trials/intervention"
     And "nciThesaurusConceptLinkPattern" is set to "https://ncit.nci.nih.gov/ncitbrowser/ConceptReport.jsp?dictionary=NCI%20Thesaurus&code=${CCODE}"
  When the user is viewing the definition with the url /def/bevacizumab
  Then an active clinical trials link to "https://www.cancer.gov/about-cancer/treatment/clinical-trials/intervention/C2039" appears after the definition text
    And an NCI Thesaurus link to "https://ncit.nci.nih.gov/ncitbrowser/ConceptReport.jsp?dictionary=NCI%20Thesaurus&code=C2039" EVS appears after the clinical trials link  
```

```gherkin
Scenario: As a user, I would like to see a link to the NCI Thesaurus entry for that drug when I am viewing a term page
  Given "dynamicListingPageBase" is set to "https://www.cancer.gov/about-cancer/treatment/clinical-trials/intervention"
     And "nciThesaurusConceptLinkPattern" is set to "https://ncit.nci.nih.gov/ncitbrowser/ConceptReport.jsp?dictionary=NCI%20Thesaurus&code=${CCODE}"
  When the user is viewing the definition with the url /def/bevacizumab
  Then an NCI Thesaurus link to "https://ncit.nci.nih.gov/ncitbrowser/ConceptReport.jsp?dictionary=NCI%20Thesaurus&code=C2039" EVS appears after the clinical trials link  
```

```gherkin
Scenario: negative: As a user, I do not see see a clinical trials or NCI Thesaurus link when it is not provided
  Given the user is viewing the definition with the url /def/801361
    And the page title is "A2A receptor antagonist EOS100850"
  Then the "Check for active clinical trials using this agent" link does not appear on the page
    And the "View this agent in the NCI Thesaurus" link does not appear on the page
```

### Resources:
* Mockups: https://publicis.invisionapp.com/share/QDYHYSLCBZY#/screens/429245338

### Notes
* See mockup for changes to the link text and display.
* The "active clinical trials link" refers to the text, `Check for active clinical trials using this agent` and the URL points to the "Intervention Dynamic Listing Page" and has the format of https://www.cancer.gov/about-cancer/treatment/clinical-trials/intervention/[C-Code or Pretty Url] where C-Code is the NCIConceptId element from the API and Pretty Url is the PrettyUrlName parameter.
  * NOTE: The pretty urls will be out of sync with the dynamic listing pages until the dynamic listing pages are moved to react. So use the C-Code for now.
* "an NCI Thesaurus link" refers to the text `View this agent on the NCI Thesaurus` which links to the corresponding NCI Thesaurus entry (URL: https://ncit.nci.nih.gov/ncitbrowser/ConceptReport.jsp?dictionary=NCI%20Thesaurus&code=[C Code] where C-Code is the NCIConceptId from the API)

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by karlikpj**
Closes #28 

* Active Clinical Trials and Thesaurus links

### CODE REVIEW

File: src/components/molecules/definitionitem/DefinitionItem.jsx [hunk 1]
```diff
@@ -4,10 +4,32 @@ import React from 'react';
...
+	const [
+		{ nciThesaurusConceptLink, ctsDynamicListingPageBase },
+	] = useStateValue();
 	const drugInfoSummaryLink = payload?.drugInfoSummaryLink;
 	const definitionText = payload.definition?.html;
+	const nciConceptId = payload?.nciConceptId;
+
+	const getLinks = () => {
+		return (
+			<span>
```
Comment (arcepaul) [inline src/components/molecules/definitionitem/DefinitionItem.jsx:19]: use a <div>, not a <span> as a wrapper

File: src/components/molecules/definitionitem/DefinitionItem.jsx [hunk 2]
```diff
@@ -4,10 +4,32 @@ import React from 'react';
...
 
 const DefinitionItem = ({ payload }) => {
+	const [
+		{ nciThesaurusConceptLink, ctsDynamicListingPageBase },
+	] = useStateValue();
 	const drugInfoSummaryLink = payload?.drugInfoSummaryLink;
 	const definitionText = payload.definition?.html;
+	const nciConceptId = payload?.nciConceptId;
+
+	const getLinks = () => {
```
Comment (arcepaul) [inline src/components/molecules/definitionitem/DefinitionItem.jsx:17]: rename to something like renderSupplementalLinks()

File: src/components/molecules/definitionitem/DefinitionItem.jsx [hunk 3]
```diff
@@ -4,10 +4,32 @@ import React from 'react';
 import './DefinitionItem.scss';
 
 import { SynonymTable } from '../../atomic';
+import { useStateValue } from '../../../store/store.js';
 
 const DefinitionItem = ({ payload }) => {
+	const [
```
Comment (arcepaul) [inline src/components/molecules/definitionitem/DefinitionItem.jsx:10]: This should be a dumb component.  Pass these values in as props from parent instead

File: src/components/molecules/definitionitem/DefinitionItem.jsx (hunk 1 above)
Comment (arcepaul) [reply src/components/molecules/definitionitem/DefinitionItem.jsx:19]: also find a class for the div to make it a css landmark

File: src/components/molecules/definitionitem/DefinitionItem.jsx [hunk 4]
```diff
@@ -4,15 +4,41 @@ import React from 'react';
...
+	aliases,
+	name,
+}) => {
+	const [
+		{ nciThesaurusConceptLink, ctsDynamicListingPageBase },
+	] = useStateValue();
 
+	const renderSupplementalLinks = () => {
+		return (
+			<div role="contentinfo">
```
Comment (arcepaul) [inline src/components/molecules/definitionitem/DefinitionItem.jsx:23]: this isn't a proper role for this.  please remove role and add `class="supplemental-links"`

**Comment by arcepaul** [comment]
package-lock is appearing as a changed file again.

File: src/components/molecules/definitionitem/DefinitionItem.jsx [hunk 5]
```diff
@@ -4,15 +4,41 @@ import React from 'react';
...
+				</a>
+				<a
+					className="nci-thesaurus-link"
+					href={`${nciThesaurusConceptLink}&code=${nciConceptId}`}>
+					View this agent in the NCI Thesaurus
+				</a>
+			</div>
+		);
+	};
+	console.log(nciConceptId);
```
Comment (arcepaul) [inline src/components/molecules/definitionitem/DefinitionItem.jsx:37]: remove

File: src/components/molecules/definitionitem/DefinitionItem.jsx (hunk 4 above)
Comment (arcepaul) [reply src/components/molecules/definitionitem/DefinitionItem.jsx]: actually, maybe this should be `class="term-definition__supplemental-links"` following BEM.  but then you would also have to look at renaming other things like, `term-definition__synonym-table`, `term-definition__term`, `term-definition__definition`, etc...

File: src/components/molecules/definitionitem/DefinitionItem.jsx [hunk 6]
```diff
@@ -23,14 +49,21 @@ const DefinitionItem = ({ payload }) => {
...
+				{nciConceptId && renderSupplementalLinks(nciConceptId)}
+				{aliases && <SynonymTable aliases={aliases} />}
 			</div>
 		</>
 	);
 };
 
 DefinitionItem.propTypes = {
 	payload: PropTypes.object,
+	drugInfoSummaryLink: PropTypes.object,
```
Comment (arcepaul) [inline src/components/molecules/definitionitem/DefinitionItem.jsx:61]: when I said to add drugInfoSummaryLink and nciConceptId as props, I didn't think everything else needed to be moved to props as well.  Maybe we could have defined a shape for payload, where things like name and termID were required.

File: src/components/molecules/definitionitem/DefinitionItem.jsx (hunk 4 above)
Comment (karlikpj) [reply src/components/molecules/definitionitem/DefinitionItem.jsx]: I'd assume this goes for search-box and other items/views..

**Review by kate-mashkina — CHANGES_REQUESTED** [review]
The only note for the acceptance tests - please add negative scenario per AC in the change request
Scenario: negative: As a user, I do not see see a clinical trials or NCI Thesaurus link when it is not provided
  Given the user navigates to "/def/801361"
    Then the page title is "A2A receptor antagonist EOS100850"
  Then the "Check for active clinical trials using this agent" link does not appear on the page
    And the "View this agent in the NCI Thesaurus" link does not appear on the page

**Review by KateMashkinaNIH — APPROVED** [review]

**Comment by karlikpj** [comment]
@VictoriaSunNIH @lburack @mworrest @blilianyu Please review
https://react-app-dev.cancer.gov/drug-dictionary-app/pr-35/def/bevacizumab

**Review by kate-mashkina — APPROVED** [review]

**Comment by blilianyu** [comment]
"View this agent..." should be on the next line
Screen Shot 2020-09-29 at 10 39 53 AM
cc: @mworrest @VictoriaSunNIH @lburack

**Comment by blilianyu** [comment]
@karlikpj New React link looks good!

@mworrest @VictoriaSunNIH @lburack

**Review by arcepaul — APPROVED** [review]

**Pull request merged**